Tools/cases_generator/optimizer_generator.py

```diff
@@ -45,7 +46,7 @@ def declare_variables(uop: Uop, out: CWriter, skip_inputs: bool) -> None:
     variables = {"unused"}
     if not skip_inputs:
         for var in reversed(uop.stack.inputs):
-            if var.name not in variables:
+            if var.name not in variables and var.used:
```

Comment:
```suggestion
            if var.used and var.name not in variables:
```

Probably slightly more efficient that way